cpp/Naive.cpp

```diff
@@ -515,18 +563,18 @@ namespace ahr {
 
     }
 
-    std::pair<Real, Real> Naive::calculateEnergies() const {
-        Real magnetic = 0, kinetic = 0;
+    Naive::Energies Naive::calculateEnergies() const {
+        Energies e;
         for_each_kxky([&](Dim kx, Dim ky) {
-            magnetic += kPerp2(kx, ky) * std::norm(momentsNew_K(kx, ky, A_PAR));
+            e.magnetic += kPerp2(kx, ky) * std::norm(moments_K(kx, ky, A_PAR));
             if (rhoI < smallRhoI) {
-                kinetic += kPerp2(kx, ky) * std::norm(phi_K_New(kx, ky));
+                e.kinetic += kPerp2(kx, ky) * std::norm(phi_K(kx, ky));
             } else {
-                kinetic -= 1.0 / (rhoI * rhoI) * (Gamma0(kPerp2(kx, ky) * rhoI * rhoI / 2.0) - 1) *
-                           std::norm(phi_K_New(kx, ky));
+                e.kinetic -= 1.0 / (rhoI * rhoI) * (Gamma0(kPerp2(kx, ky) * rhoI * rhoI / 2.0) - 1) *
+                           std::norm(phi_K(kx, ky));
             }
         });
 
-        return {magnetic, kinetic};
+        return e;
     }
 }
```

Comment:
This was extremely cursed to look through but I'm pretty sure everything looks good. From what I understand, basically just moving the m>2 case behind an if statement and setting gmin = 0 where necessary in some of the bracket calcs. 

As an aside, I think the way Viriato deals with this is that its for loops across the g's dont get executed if gmin+1 > ngtot-1, so when ngtot = 2, gmin=1 (or 2 not sure) it just doesnt activate the loop and the g's are just never updated. But then it keeps around an extra couple of zero arrays for g of size ngtot. I'm assuming the way you've done it here is better from a performance perspective.